include/boost/geometry/algorithms/detail/is_valid/box.hpp

```diff
@@ -30,15 +31,23 @@ namespace detail { namespace is_valid
 template <typename Box, std::size_t I>
 struct has_valid_corners
 {
-    static inline bool apply(Box const& box)
+    template <typename VisitPolicy>
+    static inline bool apply(Box const& box, VisitPolicy& visitor)
     {
-        if ( geometry::get<geometry::max_corner, I-1>(box)
-             <=
-             geometry::get<geometry::min_corner, I-1>(box) )
+        if (math::equals(geometry::get<geometry::min_corner, I-1>(box),
+                         geometry::get<geometry::max_corner, I-1>(box)))
         {
+            visitor.template apply<failure_wrong_dimension>();
```

Comment:
So this is invalid (was already). I find the word "dimension" a little confusing - it is the topological dimension, no? Maybe just add "topological" here?